projects/gui/src/newtournamentdialog.cpp

```diff
@@ -240,6 +254,35 @@ Tournament* NewTournamentDialog::createTournament(GameManager* gameManager) cons
 	t->setGamesPerEncounter(ui->m_gamesPerEncounterSpin->value());
 	t->setRoundMultiplier(ui->m_roundsSpin->value());
 
+	const QString& fileName = ui->m_suiteFileEdit->text();
+	if (!fileName.isEmpty())
+	{
+		OpeningSuite::Format format = OpeningSuite::PgnFormat;
+		if (fileName.endsWith(".pgn"))
+			format = OpeningSuite::PgnFormat;
+
+		OpeningSuite::Order order = OpeningSuite::SequentialOrder;
+		if (ui->m_seqOrderRadio->isChecked())
+			order = OpeningSuite::SequentialOrder;
+		else if (ui->m_randomOrderRadio->isChecked())
+			order = OpeningSuite::RandomOrder;
+
+		OpeningSuite* suite = new OpeningSuite(ui->m_suiteFileEdit->text(),
+		                         format, order, 0);
+		if (suite->initialize())
+			t->setOpeningSuite(suite);
+		else
+		{
+			delete suite;
+			return 0;
+		}
+	}
+
+	if (ui->m_repeatCheckBox->isChecked())
```

Comment:
This whole thing can be squeezed to a single line:

`t->setOpeningRepetition(ui->m_repeatCheckBox->isChecked());`